File: lambdas/personalizacion/GetCustomization.js
const AWS = require('aws-sdk');
const dynamodb = new AWS.DynamoDB.DocumentClient();

exports.handler = async (event) => {
    const { tenant_id } = event.pathParameters;

    const params = {
        TableName: 't_customizations',
        Key: { tenant_id }
    };

    try {
        const response = await dynamodb.get(params).promise();

        if (response.Item) {
            return {
                statusCode: 200,
                body: JSON.stringify(response.Item)
            };
        } else {
            return {
                statusCode: 404,
                body: JSON.stringify({ error: 'Las personalizaciones no fueron encontradas' })
            };
        }
    } catch (error) {
        console.error(error);
        return {
            statusCode: 500,
            body: JSON.stringify({ error: error.message })
        };
    }
};

File: lambdas/personalizacion/getCustomization.py
import boto3
import json

def lambda_handler(event, context):
    dynamodb = boto3.resource('dynamodb')
    table = dynamodb.Table('t_customizations')

    tenant_id = event['pathParameters']['tenant_id']

    try:
        response = table.get_item(Key={'tenant_id': tenant_id})
        if 'Item' in response:
            return {
                'statusCode': 200,
                'body': json.dumps(response['Item'])
            }
        else:
            return {
                'statusCode': 404,
                'body': json.dumps({'error': 'Las personalizaciones no fueron encontradas'})
            }
    except Exception as e:
        print(e)
        return {
            'statusCode': 500,
            'body': json.dumps({'error': str(e)})
        }
